lib/srv/alpnproxy/auth/auth_proxy_test.go

```diff
@@ -51,11 +54,13 @@ func TestDialLocalAuthServerAvailableServers(t *testing.T) {
 	authServers[0] = socket.Addr().String()
 	// multiple invalid servers to minimize chance that we select good one first try
 	for i := 0; i < 10; i++ {
-		authServers = append(authServers, "0.0.0.0:3025")
+		authServers = append(authServers, fmt.Sprintf("203.0.113.%d:3025", i+1))
 	}
 	s := NewAuthProxyDialerService(nil /* reverseTunnelServer */, "clustername", authServers, nil, nil)
 	require.Eventually(t, func() bool {
-		conn, err := s.dialLocalAuthServer(context.Background())
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
```

Comment:
Should we give ourselves a little more than 10ms?
```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), 50*time.Millisecond)
```